### PR TITLE
Add flow types to components; realted to #24

### PR DESCRIPTION
**Pull request by tricinel**
I've added some more flow types. As usual, feedback welcome.

### CODE REVIEW

File: src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js [hunk 1]
```diff
@@ -23,6 +36,9 @@ class StickyResponsiveSidebar extends Component {
     this._closeNavMenu = this._closeNavMenu.bind(this);
   }
 
+  _openNavMenu: Function;
+  _closeNavMenu: Function;
+
```
Comment (bvaughn) [inline src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js]: Maybe we should just replace these lines:
```js
    this._openNavMenu = this._openNavMenu.bind(this);
    this._closeNavMenu = this._closeNavMenu.bind(this);
  }

  _openNavMenu: Function;
  _closeNavMenu: Function;

  _openNavMenu() {
    this.setState({open: !this.state.open});
  }

  _closeNavMenu() {
    this.setState({open: false});
  }
```

With this?

File: flow-typed/glamor.js [hunk 2]
```diff
@@ -5,3 +5,13 @@ declare module 'glamor' {
...
 }
+
+declare module 'glamor/react' {
+  declare module.exports: {
+    createElement: any,
+    dom: any,
+    vars: any,
+    makeTheme: any,
+    propMerge: Function,
+  };
```
Comment (bvaughn) [inline flow-typed/glamor.js]: I wouldn't expect to see anything here beyond `createElement`. Are these others necessary?

Comment (tricinel) [reply flow-typed/glamor.js]: I was actually just looking at the Typescript typings available in glamor: https://github.com/threepointone/glamor/blob/master/react.d.ts.

File: src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js (hunk 1 above)
Comment (tricinel) [reply src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js]: Yes, absolutely. I didn't want to modify the actual code too much.

Comment (bvaughn) [reply src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js]: Friendly ping.

Comment (tricinel) [reply src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js]: Hi, truly sorry! I've been a bit swamped, but I have a few more things I've done so I'll push those ASAP (probably Monday).

**Comment by tricinel** [comment]
@bvaughn, I've actually left some comments here and there. I see there's a conflict in ErrorDecoder, so no worries I'll fix that as well in the next push (should be tomorrow).

**Comment by tricinel** [comment]
@bvaughn, how are we looking? I have a feeling I should have done a rebase instead of merge...What do you think?

**Comment by bvaughn** [comment]
No big deal WRT rebase vs merge. When we merge PRs into the main repo here, we squash anyway.

As for how we're looking- haven't had time to review yet. Will try to find time...sometime soon.

**Comment by tricinel** [comment]
Alright, sounds good! No worries about reviewing. I have a few more changes that I've already done, but not committed because I wasn't sure about merge vs rebase.

Will polish that up and push then.

Cheers!

File: src/layouts/index.js [hunk 3]
```diff
@@ -2,10 +2,13 @@
  * Copyright (c) 2013-present, Facebook, Inc.
  *
  * @emails react-core
+ * @flow
  */
 
 'use strict';
 
+// FIXME: I actually cannot figure out where this Template is used
```
Comment (bvaughn) [inline src/layouts/index.js]: This is a built-in Gatsby thing: https://www.gatsbyjs.org/docs/building-with-components/#layout-components

File: src/layouts/index.js [hunk 4]
```diff
@@ -23,7 +24,12 @@ import 'glamor/reset';
 import 'css/reset.css';
 import 'css/algolia.css';
 
-class Template extends Component {
+type Props = {
+  children: any, // I don't think this is a Node type
```
Comment (bvaughn) [inline src/layouts/index.js]: This is a Function. (Also a built-in Gatsby thing): https://www.gatsbyjs.org/docs/building-with-components/#layout-components

File: src/pages/blog/all.html.js [hunk 5]
```diff
@@ -76,18 +76,22 @@ const AllBlogPosts = ({data}: Props) => (
...
+                  }}>
+                  by{' '}
+                  {toCommaSeparatedList(node.frontmatter.author, author => (
+                    <span key={author.frontmatter.name}>
+                      {author.frontmatter.name}
+                    </span>
+                  ))}
+                </div>
+              ) : (
+                <div />
```
Comment (bvaughn) [inline src/pages/blog/all.html.js]: Probably don't need the empty `<div>`? Could just do `null`?

**Comment by bvaughn** [comment]
This looks great! Sorry for getting distracted and taking forever to circle back to this PR. 😄  Thanks so much!

**Comment by tricinel** [comment]
No problem! Thanks for reviewing! A few more coming down the pipe this next week. We're nearly there :)

**Pull request merged**